lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -695,4 +746,41 @@ private Function<Integer, Object> getConverter(PrimitiveType primitiveType)
             super(columnPath, Integer.class);
         }
     }
+
+    /**
+     * Check if the predicateValue might be in the bloomfilter
+     *
+     * @param bloomFilter parquet bloomfilter.
+     * @param predicateValue effective discrete predicate value.
+     * @param sqlType Type that contains information about the type schema from connector's metadata
+     * @return true if the predicateValue might be in the bloomfilter, false if the predicateValue absolutely is not in the bloomfilter
+     */
+    @VisibleForTesting
+    public static boolean checkInBloomFilter(BloomFilter bloomFilter, Object predicateValue, Type sqlType)
+    {
+        // todo: support TIMESTAMP, and DECIMAL
+        if (sqlType == TINYINT || sqlType == SMALLINT || sqlType == INTEGER || sqlType == BIGINT || sqlType == DATE) {
+            return bloomFilter.findHash(bloomFilter.hash((int) asLong(predicateValue)));
+        }
+        else if (sqlType == DOUBLE) {
+            return bloomFilter.findHash(bloomFilter.hash(predicateValue));
+        }
+        else if (sqlType == REAL) {
+            return bloomFilter.findHash(bloomFilter.hash(intBitsToFloat(toIntExact(((Number) predicateValue).intValue()))));
+        }
+        else if (sqlType instanceof VarcharType || sqlType instanceof CharType || sqlType instanceof VarbinaryType || sqlType instanceof UuidType) {
```

Comment:
I'm not sure that the implementation is correct for `CharType`
We might need to use `Chars#padSpaces`. Please verify

Reply:
sure, will verify and add test case

Reply:
Current implementation might be correct if the bloom filter hash written by writer as well as the Slice object coming from the predicate are unpadded values